tempodb/encoding/vparquet3/compactor.go

```diff
@@ -146,6 +147,18 @@ func (c *Compactor) Compact(ctx context.Context, l log.Logger, r backend.Reader,
 			return nil, fmt.Errorf("error iterating input blocks: %w", err)
 		}
 
+		// ship block to backend if done
+		if currentBlock != nil && cmd.CutBlock(currentBlock.meta, lowestID) {
+			currentBlockPtrCopy := currentBlock
+			currentBlockPtrCopy.meta.StartTime = minBlockStart
```

Comment:
Hmm this block was moved from the bottom of the loop to the top so that the `CutBlock` callback can inspect the next trace (`lowestID`).  You're right it does seem to be out of date and the copy is no longer needed.  I'll take a look.